src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1107,19 +1147,25 @@
 
 (declare push-down-context)
 
+(defn unsupported-fields
+  [field allowed-fields]
+  (let [supported-fns ["count"]]
+    (remove (set (concat allowed-fields (map #(vector "function" %) supported-fns)))
```

Comment:
So we already have a function that does this in scf/storage.clj called basic-diff, any way we could move that to utils and use that?